models/model_upload/multimodal_models/vllm-miniCPM-2.6/1/model.py

```diff
@@ -0,0 +1,259 @@
+    models = self.client.models.list()
+    self.model = models.data[0].id
+
+  def predict(self,
+              request: service_pb2.PostModelOutputsRequest) -> service_pb2.MultiOutputResponse:
+    """This is the method that will be called when the runner is run. It takes in an input and
+    returns an output.
+    """
+
+    inference_params = get_inference_params(request)
+
+    streams = []
+    for input in request.inputs:
+
+      # it contains the input data for the model
+      input_data = input.data
+
+      stream = stream_completion(self.model, self.client, input_data, inference_params)
+
+      streams.append(stream)
+
+    outputs = [resources_pb2.Output() for _ in request.inputs]
+    for output in outputs:
+      output.status.code = status_code_pb2.SUCCESS
+
+    for chunk_batch in itertools.zip_longest(*streams, fillvalue=None):
+      for idx, chunk in enumerate(chunk_batch):
+        outputs[idx].data.text.raw += chunk.choices[0].delta.content if (
+            chunk and chunk.choices[0].delta.content) is not None else ''
+
+    return service_pb2.MultiOutputResponse(outputs=outputs,)
+
+  def generate(self, request: service_pb2.PostModelOutputsRequest
+              ) -> Iterator[service_pb2.MultiOutputResponse]:
+    """Example yielding a whole batch of streamed stuff back."""
+
+    # Get the inference params from the request
+    inference_params = get_inference_params(request)
+
+    streams = []
+    for input in request.inputs:
+
+      # it contains the input data for the model
+      input_data = input.data
+
+      stream = stream_completion(self.model, self.client, input_data, inference_params)
+
+      streams.append(stream)
+    for chunk_batch in itertools.zip_longest(*streams, fillvalue=None):
+      resp = service_pb2.MultiOutputResponse()
+
+      for chunk in chunk_batch:
+        output = resp.outputs.add()
+        output.data.text.raw = (chunk.choices[0].delta.content
+                                if (chunk and chunk.choices[0].delta.content) is not None else '')
+        output.status.code = status_code_pb2.SUCCESS
+      yield resp
+
+  def stream(self, request_iterator: Iterator[service_pb2.PostModelOutputsRequest]
+            ) -> Iterator[service_pb2.MultiOutputResponse]:
+    NotImplementedError("Stream method is not implemented for the models.")
+
+
+if __name__ == '__main__':
+  # Make sure you set these env vars before running the example.
+  # CLARIFAI_PAT
+  # CLARIFAI_USER_ID
+  # CLARIFAI_API_BASE
+  # CLARIFAI_RUNNER_ID
+  # CLARIFAI_NODEPOOL_ID
+  # CLARIFAI_COMPUTE_CLUSTER_ID
+
+  # You need to first create a runner in the Clarifai API and then use the ID here.
+  MyRunner(
+      runner_id=os.environ["CLARIFAI_RUNNER_ID"],
+      nodepool_id=os.environ["CLARIFAI_NODEPOOL_ID"],
+      compute_cluster_id=os.environ["CLARIFAI_COMPUTE_CLUSTER_ID"],
+      base_url=os.environ["CLARIFAI_API_BASE"],
+      num_parallel_polls=int(os.environ.get("CLARIFAI_NUM_THREADS", 1)),
+  ).start()
```

Comment:
This part is difficult to understand. It seems inconsistent to other examples. 

Why not include a .predict and a .generate example here so folks can test with multimodality before uploading. Currently test-locally will catch an llm input, but there is no way for the user to test multimodality before uploading.

Reply:
I removed this part now.

I don't understand what you meant by  `include a .predict and a .generate example here`. btw `test-locally` works with multimodal models too, it test the model on sample image and text input